internal/controllers/synthesis/lifecycle.go

```diff
@@ -91,13 +94,16 @@ func (c *podLifecycleController) Reconcile(ctx context.Context, req ctrl.Request
 	if exists {
 		// The pod is still running.
 		// Poll periodically to check if has timed out.
+		if syn.Spec.PodTimeout == nil {
+			return ctrl.Result{}, nil
+		}
 		return ctrl.Result{RequeueAfter: syn.Spec.PodTimeout.Duration}, nil
 	}
 
 	if comp.DeletionTimestamp != nil {
 		// If the composition was being synthesized at the time of deletion we need to swap the previous
 		// state back to current. Otherwise we'll get stuck waiting for a synthesis that can't happen.
-		if comp.Status.CurrentSynthesis == nil || comp.Status.CurrentSynthesis.Synthesized == nil {
+		if comp.Status.PreviousSynthesis != nil && (comp.Status.CurrentSynthesis == nil || comp.Status.CurrentSynthesis.Synthesized == nil) {
```

Comment:
I'm not sure this is necessary, but seems more sane. Don't swap back if there isn't anything to swap to.

Reply:
Hmm I think this might fix deletion during the initial synthesis, since otherwise we'd keep swapping back to previous for every tick of the loop. Made a note to add a test for this scenario.